File: MCUSB4sd-DD13/MCUSB4sd-DD13/Samples/CS_NET/Sample1/Form1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using HpmcstdCs;

namespace Sample1
{
	/// <summary>
	/// Form1 の概要の説明です。
	///	１.「Board open」ボタンでボードをオープンします。例としてスイッチ番号０のボードをオープンします。
	/// ２.「Board set up」ボタンで第１軸の各種設定を行います。
	/// ３.「Drive start」ボタンで第１軸のドライブを開始します。
	/// ４.「Status read」ボタンで第１軸の各種状態を表示します。
	/// ５.「Board close」ボタンでボードをクローズします。
	/// </summary>
	public class Form1 : System.Windows.Forms.Form
	{
		private System.Windows.Forms.Button button1;
		private System.Windows.Forms.Button button2;
		private System.Windows.Forms.Button button3;
		private System.Windows.Forms.Button button4;
		private System.Windows.Forms.Button button5;
		/// <summary>
		/// 必要なデザイナ変数です。
		/// </summary>
		private System.ComponentModel.Container components = null;

		//
		string szBoardName = "MCUSB4sd";
		public const ushort MCUSB4sd_AXIS1 = 0;
		public const ushort MCUSB4sd_AXIS2 = 1;
		public const ushort MCUSB4sd_AXIS3 = 2;
		public const ushort MCUSB4sd_AXIS4 = 3;
		public const int WM_USER = 0x400;
		public uint hDevice;

		public Form1()
		{
			//
			// Windows フォーム デザイナ サポートに必要です。
			//
			InitializeComponent();

			//
			// TODO: InitializeComponent 呼び出しの後に、コンストラクタ コードを追加してください。
			//
		}

		/// <summary>
		/// 使用されているリソースに後処理を実行します。
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if (components != null) 
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows フォーム デザイナで生成されたコード 
		/// <summary>
		/// デザイナ サポートに必要なメソッドです。このメソッドの内容を
		/// コード エディタで変更しないでください。
		/// </summary>
		private void InitializeComponent()
		{
			this.button1 = new System.Windows.Forms.Button();
			this.button2 = new System.Windows.Forms.Button();
			this.button3 = new System.Windows.Forms.Button();
			this.button4 = new System.Windows.Forms.Button();
			this.button5 = new System.Windows.Forms.Button();
			this.SuspendLayout();
			// 
			// button1
			// 
			this.button1.Location = new System.Drawing.Point(24, 24);
			this.button1.Name = "button1";
			this.button1.Size = new System.Drawing.Size(88, 24);
			this.button1.TabIndex = 0;
			this.button1.Text = "Board open";
			this.button1.Click += new System.EventHandler(this.button1_Click);
			// 
			// button2
			// 
			this.button2.Location = new System.Drawing.Point(24, 64);
			this.button2.Name = "button2";
			this.button2.Size = new System.Drawing.Size(88, 24);
			this.button2.TabIndex = 1;
			this.button2.Text = "Board close";
			this.button2.Click += new System.EventHandler(this.button2_Click);
			// 
			// button3
			// 
			this.button3.Location = new System.Drawing.Point(24, 104);
			this.button3.Name = "button3";
			this.button3.Size = new System.Drawing.Size(88, 24);
			this.button3.TabIndex = 2;
			this.button3.Text = "Board set up";
			this.button3.Click += new System.EventHandler(this.button3_Click);
			// 
			// button4
			// 
			this.button4.Location = new System.Drawing.Point(24, 144);
			this.button4.Name = "button4";
			this.button4.Size = new System.Drawing.Size(88, 24);
			this.button4.TabIndex = 3;
			this.button4.Text = "Drive start";
			this.button4.Click += new System.EventHandler(this.button4_Click);
			// 
			// button5
			// 
			this.button5.Location = new System.Drawing.Point(24, 184);
			this.button5.Name = "button5";
			this.button5.Size = new System.Drawing.Size(88, 24);
			this.button5.TabIndex = 4;
			this.button5.Text = "Status read";
			this.button5.Click += new System.EventHandler(this.button5_Click);
			// 
			// Form1
			// 
			this.AutoScaleBaseSize = new System.Drawing.Size(5, 12);
			this.ClientSize = new System.Drawing.Size(136, 224);
			this.Controls.Add(this.button5);
			this.Controls.Add(this.button4);
			this.Controls.Add(this.button3);
			this.Controls.Add(this.button2);
			this.Controls.Add(this.button1);
			this.Name = "Form1";
			this.StartPosition = System.Windows.Forms.FormStartPosition.CenterScreen;
			this.Text = "Form1";
			this.ResumeLayout(false);

		}
		#endregion

		/// <summary>
		/// アプリケーションのメイン エントリ ポイントです。
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Application.Run(new Form1());
		}

		///////////////////////////////////////////////////////////////////////////
		// Board open
		///////////////////////////////////////////////////////////////////////////
		private void button1_Click(object sender, System.EventArgs e)
		{
			// ボードをオープンします。例としてスイッチ番号０のボードをオープンします。
			hDevice = Hpmcstd.McsdOpen( szBoardName, 0);
			if ( 0xFFFFFFFF == hDevice ) 
			{
				MessageBox.Show( "ボードのオープンに失敗しました\n", szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Error);
			}
			else
			{
				MessageBox.Show( "Successful Device Open\n", szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Information);
			}
		}

		///////////////////////////////////////////////////////////////////////////
		// Board close
		///////////////////////////////////////////////////////////////////////////
		private void button2_Click(object sender, System.EventArgs e)
		{
			uint	dwRet = Hpmcstd.McsdClose(hDevice);
			if ( dwRet==0 ) 
			{
				MessageBox.Show( "Device Close\n", szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Information);
			}
			else
			{
				string szStr = String.Format(
					"{0} error\n\n" + 
					"Error code: {1:D04}", szBoardName, dwRet);
				MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Error);
			}
		}

		///////////////////////////////////////////////////////////////////////////
		// Drive start
        //	第１軸の設定を行います。
		///////////////////////////////////////////////////////////////////////////
		private void button3_Click(object sender, System.EventArgs e)
		{
			uint	dwRet;
			string	szStr;

			Hpmcstd.MCSDSPDDATAPPS	SpeedPPS = new Hpmcstd.MCSDSPDDATAPPS();

			// パルス出力方式	２パルス方式
			// DIR   出力端子 	CWパルス  アクティブ Hi
			// PULSE 出力端子	CCWパルス アクティブ Hi
			dwRet = Hpmcstd.McsdSetPulseMode( hDevice, MCUSB4sd_AXIS1, 4);
			if ( dwRet!=0 ) 
			{
				szStr = String.Format( "{0} error\n\nError code: {1:D04}", szBoardName, dwRet);
				MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Error);
				return;
			}

			// +LMT		  入力信号アクティブレベル	Low
			// -LMT		  入力信号アクティブレベル	Low
			// ALARM	  入力信号アクティブレベル	Low
			// INPOSITION 入力信号アクティブレベル	Low
			dwRet = Hpmcstd.McsdSetLimit( hDevice, MCUSB4sd_AXIS1, 0x33);
			if ( dwRet!=0 ) 
			{
				szStr = String.Format( "{0} error\n\nError code: {1:D04}", szBoardName, dwRet);
				MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Error);
				return;
			}

			// EXTERNAL COUNTER 入力仕様	２相信号４逓倍入力
			dwRet = Hpmcstd.McsdSetExternalCounterMode( hDevice, MCUSB4sd_AXIS1, 3);
			if ( dwRet!=0 ) 
			{
				szStr = String.Format( "{0} error\n\nError code: {1:D04}", szBoardName, dwRet);
				MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Error);
				return;
			}

			// Alarm stop mode set
			// Inposition wait mode reset
			// Limit input stop enable (Emergency stop)
			dwRet = Hpmcstd.McsdSetSignalStop( hDevice, MCUSB4sd_AXIS1, 1, 0, 2);
			if ( dwRet!=0 ) 
			{
				szStr = String.Format( "{0} error\n\nError code: {1:D04}", szBoardName, dwRet);
				MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Error);
				return;
			}

			// 自起動速度	100PPS
			// 最高速度 	5000PPS
			// 加速時間 	250mSec
			// Ｓ字加減率	0%（直線加減速）
			SpeedPPS.dAccel = 250;
			SpeedPPS.dHighSpeed = 5000;
			SpeedPPS.dLowSpeed = 100;
			SpeedPPS.dSratio = 0;
			dwRet = Hpmcstd.McsdSetSpeedPPS( hDevice, MCUSB4sd_AXIS1, ref SpeedPPS);
			if ( dwRet!=0 ) 
			{
				szStr = String.Format( "{0} error\n\nError code: {1:D04}", szBoardName, dwRet);
				MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Error);
				return;
			}

			// カウンタ設定
			//
			// INTERNAL COUNTER 及び EXTERNAL COUNTER に 0h を書き込みます
			dwRet = Hpmcstd.McsdSetCounter( hDevice, MCUSB4sd_AXIS1, 0, 0);
			if ( dwRet!=0 ) 
			{
				szStr = String.Format( "{0} error\n\nError code: {1:D04}", szBoardName, dwRet);
				MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Error);
				return;
			}
			dwRet = Hpmcstd.McsdSetCounter( hDevice, MCUSB4sd_AXIS1, 1, 0);
			if ( dwRet!=0 ) 
			{
				szStr = String.Format( "{0} error\n\nError code: {1:D04}", szBoardName, dwRet);
				MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Error);
				return;
			}

			// コンパレータ設定
			//
			// INTERNAL COMPARATER 及び EXTERNAL COMPARATER に 0h を書き込みます
			dwRet = Hpmcstd.McsdSetComparator( hDevice, MCUSB4sd_AXIS1, 0, 0);
			if ( dwRet!=0 ) 
			{
				szStr = String.Format( "{0} error\n\nError code: {1:D04}", szBoardName, dwRet);
				MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Error);
				return;
			}
			dwRet = Hpmcstd.McsdSetComparator( hDevice, MCUSB4sd_AXIS1, 1, 0);
			if ( dwRet!=0 ) 
			{
				szStr = String.Format( "{0} error\n\nError code: {1:D04}", szBoardName, dwRet);
				MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Error);
				return;
			}

			//
			MessageBox.Show( "第１軸の設定が完了しました\n", szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Information);
		}

		///////////////////////////////////////////////////////////////////////////
		// Drive start
		//	第１軸のドライブを行います。
		///////////////////////////////////////////////////////////////////////////
		private void button4_Click(object sender, System.EventArgs e)
		{
			uint	dwRet;
			string	szStr;

			// 第１軸を＋方向に 10000 パルスドライブします。
			dwRet = Hpmcstd.McsdDriveStart( hDevice, MCUSB4sd_AXIS1, 0, 10000 );
			if ( dwRet!=0 ) 
			{
				szStr = String.Format( "{0} error\n\nError code: {1:D04}", szBoardName, dwRet);
				MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Error);
				return;
			}
			szStr = String.Format( "{0}\n" + "制御軸 {1} ＋方向に 10000 パルス出力しています", szBoardName, MCUSB4sd_AXIS1 );
			MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Information);
		}

		///////////////////////////////////////////////////////////////////////////
		// status read
		//	第１軸の各種情報を表示します。
		///////////////////////////////////////////////////////////////////////////
		private void button5_Click(object sender, System.EventArgs e)
		{
			uint	dwRet;
			string	szStr;
			uint	dwData;
			Hpmcstd.MCSDSTATUS	Status	= new Hpmcstd.MCSDSTATUS();

			// 内部カウンターを表示します。
			dwRet = Hpmcstd.McsdGetCounter( hDevice, MCUSB4sd_AXIS1, 0, out dwData );
			if ( dwRet!=0 ) 
			{
				szStr = String.Format( "{0} error\n\nError code: {1:D04}", szBoardName, dwRet);
				MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Error);
				return;
			}
			// 内部カウンタは 28 ビットなので符号拡張します。
			if ( (dwData & 0x08000000) == 0x08000000 ) dwData |= 0xF0000000;
			szStr = String.Format("{0}\n" + "制御軸 {1} 内部カウンター: {2:D}", szBoardName, MCUSB4sd_AXIS1, dwData);
			MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Information);

			// 外部カウンターを表示します。
			dwRet = Hpmcstd.McsdGetCounter( hDevice, MCUSB4sd_AXIS1, 1, out dwData );
			if ( dwRet!=0 ) 
			{
				szStr = String.Format( "{0} error\n\nError code: {1:D04}", szBoardName, dwRet);
				MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Error);
				return;
			}
			// 外部カウンタは 28 ビットなので符号拡張します。
			if ( (dwData & 0x08000000) == 0x08000000 ) dwData |= 0xF0000000;
			szStr = String.Format("{0}\n" + "制御軸 {1} 外部カウンター: {2:D}", szBoardName, MCUSB4sd_AXIS1, dwData);
			MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Information);

			// エンドステータスを取得して表示します。
			// メカニカルシグナルを取得して表示します。
			// ユニバーサルシグナルを取得して表示します。
			dwRet = Hpmcstd.McsdGetStatus( hDevice, MCUSB4sd_AXIS1, out Status);
			if ( dwRet!=0 ) 
			{
				szStr = String.Format( "{0} error\n\nError code: {1:D04}", szBoardName, dwRet);
				MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Error);
				return;
			}
			szStr = String.Format(
				"{0}\n" + "制御軸 {1}\n\n" +
				"エンドステータス : {2:X02}h\n" +
				"メカニカルシグナル : {3:X02}h\n" +
				"ユニバーサルシグナル : {4:X02}h\n"
				, szBoardName, MCUSB4sd_AXIS1, Status.bEnd, Status.bMechanical, Status.bUniversal);
			MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Information);

			// ドライブ中であれば減速停止します。
			if ( (Status.bDrive & 0x01) == 0x01 ) 
			{
				dwRet = Hpmcstd.McsdDriveStop( hDevice, MCUSB4sd_AXIS1, 0);
				if ( dwRet!=0 ) 
				{
					szStr = String.Format( "{0} error\n\nError code: {1:D04}", szBoardName, dwRet);
					MessageBox.Show( szStr, szBoardName, MessageBoxButtons.OK, MessageBoxIcon.Error);
					return;
				}
			}
		}
	}
}

File: MCUSB4sd-DD13/MCUSB4sd-DD13/Samples/VB_NET/Sample1/Form1.vb
Option Strict Off
Option Explicit On
Friend Class Form1
	Inherits System.Windows.Forms.Form
#Region "Windows フォーム デザイナによって生成されたコード"
	Public Sub New()
		MyBase.New()
		If m_vb6FormDefInstance Is Nothing Then
			If m_InitializingDefInstance Then
				m_vb6FormDefInstance = Me
			Else
				Try 
					'スタートアップ フォームについては、最初に作成されたインスタンスが既定インスタンスになります。
					If System.Reflection.Assembly.GetExecutingAssembly.EntryPoint.DeclaringType Is Me.GetType Then
						m_vb6FormDefInstance = Me
					End If
				Catch
				End Try
			End If
		End If
		'この呼び出しは、Windows フォーム デザイナで必要です。
		InitializeComponent()
	End Sub
	'Form は、コンポーネント一覧に後処理を実行するために dispose をオーバーライドします。
	Protected Overloads Overrides Sub Dispose(ByVal Disposing As Boolean)
		If Disposing Then
			If Not components Is Nothing Then
				components.Dispose()
			End If
		End If
		MyBase.Dispose(Disposing)
	End Sub
	'Windows フォーム デザイナで必要です。
	Private components As System.ComponentModel.IContainer
	Public ToolTip1 As System.Windows.Forms.ToolTip
    Public WithEvents Command4 As System.Windows.Forms.Button
	Public WithEvents Command3 As System.Windows.Forms.Button
	Public WithEvents Command2 As System.Windows.Forms.Button
	Public WithEvents Command1 As System.Windows.Forms.Button
	'メモ : 以下のプロシージャは Windows フォーム デザイナで必要です。
	'Windows フォーム デザイナを使って変更できます。
	'コード エディタを使って修正しないでください。
	<System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
        Me.components = New System.ComponentModel.Container
        Me.ToolTip1 = New System.Windows.Forms.ToolTip(Me.components)
        Me.Command4 = New System.Windows.Forms.Button
        Me.Command3 = New System.Windows.Forms.Button
        Me.Command2 = New System.Windows.Forms.Button
        Me.Command1 = New System.Windows.Forms.Button
        Me.SuspendLayout()
        '
        'Command4
        '
        Me.Command4.BackColor = System.Drawing.SystemColors.Control
        Me.Command4.Cursor = System.Windows.Forms.Cursors.Default
        Me.Command4.ForeColor = System.Drawing.SystemColors.ControlText
        Me.Command4.Location = New System.Drawing.Point(16, 144)
        Me.Command4.Name = "Command4"
        Me.Command4.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.Command4.Size = New System.Drawing.Size(105, 25)
        Me.Command4.TabIndex = 3
        Me.Command4.Text = "Command4"
        '
        'Command3
        '
        Me.Command3.BackColor = System.Drawing.SystemColors.Control
        Me.Command3.Cursor = System.Windows.Forms.Cursors.Default
        Me.Command3.ForeColor = System.Drawing.SystemColors.ControlText
        Me.Command3.Location = New System.Drawing.Point(16, 104)
        Me.Command3.Name = "Command3"
        Me.Command3.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.Command3.Size = New System.Drawing.Size(105, 25)
        Me.Command3.TabIndex = 2
        Me.Command3.Text = "Command3"
        '
        'Command2
        '
        Me.Command2.BackColor = System.Drawing.SystemColors.Control
        Me.Command2.Cursor = System.Windows.Forms.Cursors.Default
        Me.Command2.ForeColor = System.Drawing.SystemColors.ControlText
        Me.Command2.Location = New System.Drawing.Point(16, 64)
        Me.Command2.Name = "Command2"
        Me.Command2.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.Command2.Size = New System.Drawing.Size(105, 25)
        Me.Command2.TabIndex = 1
        Me.Command2.Text = "Command2"
        '
        'Command1
        '
        Me.Command1.BackColor = System.Drawing.SystemColors.Control
        Me.Command1.Cursor = System.Windows.Forms.Cursors.Default
        Me.Command1.ForeColor = System.Drawing.SystemColors.ControlText
        Me.Command1.Location = New System.Drawing.Point(16, 24)
        Me.Command1.Name = "Command1"
        Me.Command1.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.Command1.Size = New System.Drawing.Size(105, 25)
        Me.Command1.TabIndex = 0
        Me.Command1.Text = "Command1"
        '
        'Form1
        '
        Me.AutoScaleBaseSize = New System.Drawing.Size(5, 12)
        Me.BackColor = System.Drawing.SystemColors.Control
        Me.ClientSize = New System.Drawing.Size(137, 190)
        Me.Controls.Add(Me.Command4)
        Me.Controls.Add(Me.Command3)
        Me.Controls.Add(Me.Command2)
        Me.Controls.Add(Me.Command1)
        Me.Cursor = System.Windows.Forms.Cursors.Default
        Me.Location = New System.Drawing.Point(4, 23)
        Me.Name = "Form1"
        Me.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterScreen
        Me.Text = "Form1"
        Me.ResumeLayout(False)

    End Sub
#End Region 
#Region "アップグレード ウィザードのサポート コード"
	Private Shared m_vb6FormDefInstance As Form1
	Private Shared m_InitializingDefInstance As Boolean
	Public Shared Property DefInstance() As Form1
		Get
			If m_vb6FormDefInstance Is Nothing OrElse m_vb6FormDefInstance.IsDisposed Then
				m_InitializingDefInstance = True
				m_vb6FormDefInstance = New Form1()
				m_InitializingDefInstance = False
			End If
			DefInstance = m_vb6FormDefInstance
		End Get
		Set
			m_vb6FormDefInstance = Value
		End Set
	End Property
#End Region 
	
	Private Sub Form1_Load(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles MyBase.Load
		'Handle get
		ghW = Me.Handle.ToInt32
		'Axis
        gwAxis = MCUSB4sd_AXIS1
		'Caption
		Command1.Text = "Board open"
		Command2.Text = "Board close"
		Command3.Text = "Drive start"
		Command4.Text = "Status read"
    End Sub

    Private Sub Command1_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles Command1.Click
        Dim SpdDataPPS As MCSDSPDDATAPPS
        Dim dwErr As Integer
        'Board Open
        ghDevice = McsdOpen("MCUSB4sd", 0)
        If (ghDevice = G_FALSE) Then
            MsgBox("McsdOpen" & Chr(10) & "BOARD : MCUSB4sd" & Chr(10) & "ID : 0" & Chr(10) & "Open Error !!", MsgBoxStyle.OKOnly + MsgBoxStyle.Critical)
            Exit Sub
        End If
        MsgBox("Board Open (ID 0)", MsgBoxStyle.OKOnly + MsgBoxStyle.Information)
        'Parameter set
        '  Set pulse mode
        '    Pulse mode : 2 pulse
        '    nDIR       : CW  pulse, Active Hi
        '    nPULSE     : CCW pulse, Active Hi
        dwErr = McsdSetPulseMode(ghDevice, gwAxis, 4)
        If (dwErr) Then
            MsgBox("McsdSetPulseMode" & Chr(10) & "Error code : " & CStr(dwErr), MsgBoxStyle.OKOnly + MsgBoxStyle.Critical)
            Exit Sub
        End If
        'Parameter set
        '  Signal active level set
        '    n+LMT active level        Low
        '    n-LMT active level        Low
        '    nAlarm active level       Low
        '    nInposition active level  Low
        dwErr = McsdSetLimit(ghDevice, gwAxis, &H33S)
        If (dwErr) Then
            MsgBox("McsdSetLimit" & Chr(10) & "Error code : " & CStr(dwErr), MsgBoxStyle.OKOnly + MsgBoxStyle.Critical)
            Exit Sub
        End If
        'Parameter set
        '  External counter mode set
        '    Up/down signal count
        dwErr = McsdSetExternalCounterMode(ghDevice, gwAxis, 0)
        If (dwErr) Then
            MsgBox("McsdSetExternalCounterMode" & Chr(10) & "Error code : " & CStr(dwErr), MsgBoxStyle.OKOnly + MsgBoxStyle.Critical)
            Exit Sub
        End If
        'Parameter set
        '  Signal stop set
        '    Alarm stop mode set
        '    Inposition wait mode reset
        '    Limit input stop enable (Emergency stop)
        dwErr = McsdSetSignalStop(ghDevice, gwAxis, 1, 0, 2)
        If (dwErr) Then
            MsgBox("McsdSetSignalStop" & Chr(10) & "Error code : " & CStr(dwErr), MsgBoxStyle.OKOnly + MsgBoxStyle.Critical)
            Exit Sub
        End If
        'Counter clear
        dwErr = McsdSetCounter(ghDevice, gwAxis, 0, 0)
        If (dwErr) Then
            MsgBox("McsdSetCounter" & Chr(10) & "Error code : " & CStr(dwErr), MsgBoxStyle.OKOnly + MsgBoxStyle.Critical)
            Exit Sub
        End If
        dwErr = McsdSetCounter(ghDevice, gwAxis, 1, 0)
        If (dwErr) Then
            MsgBox("McsdSetCounter" & Chr(10) & "Error code : " & CStr(dwErr), MsgBoxStyle.OKOnly + MsgBoxStyle.Critical)
            Exit Sub
        End If
        'Speed set
        SpdDataPPS.dAccel = 250
        SpdDataPPS.dHighSpeed = 5000
        SpdDataPPS.dLowSpeed = 100
        SpdDataPPS.dSratio = 0
        dwErr = McsdSetSpeedPPS(ghDevice, gwAxis, SpdDataPPS)
        If (dwErr) Then
            MsgBox("McsdSetSpeedPPS" & Chr(10) & "Error code : " & CStr(dwErr), MsgBoxStyle.OKOnly + MsgBoxStyle.Critical)
            Exit Sub
        End If
    End Sub

    Private Sub Command2_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles Command2.Click
        Dim dwErr As Integer
        'Board Close
        dwErr = McsdClose(ghDevice)
        If (dwErr) Then
            MsgBox("McsdClose" & Chr(10) & "Error code : " & CStr(dwErr), MsgBoxStyle.OKOnly + MsgBoxStyle.Critical)
            Exit Sub
        End If
        MsgBox("Board Close (ID 0)", MsgBoxStyle.OKOnly + MsgBoxStyle.Information)
    End Sub

    Private Sub Command3_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles Command3.Click
        Dim dwErr As Integer
        'Drive Start
        '  + Index pulse drive
        dwErr = McsdDriveStart(ghDevice, gwAxis, 0, 10000)
        If (dwErr) Then
            MsgBox("McsdDriveStart" & Chr(10) & "Error code : " & CStr(dwErr), MsgBoxStyle.OKOnly + MsgBoxStyle.Critical)
            Exit Sub
        End If
        '
        MsgBox("制御軸 " & CStr(gwAxis) & " ＋方向に 10000 パルス出力しています", MsgBoxStyle.OKOnly + MsgBoxStyle.Information)
    End Sub

    Private Sub Command4_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles Command4.Click
        Dim Status As MCSDSTATUS
        Dim dwCount As Integer
        Dim dwErr As Integer
        'Status Read
        dwErr = McsdGetStatus(ghDevice, gwAxis, Status)
        If (dwErr) Then
            MsgBox("McsdGetStatus" & Chr(10) & "Error code : " & CStr(dwErr), MsgBoxStyle.OKOnly + MsgBoxStyle.Critical)
            Exit Sub
        End If
        'Internal counter read
        dwErr = McsdGetCounter(ghDevice, gwAxis, 0, dwCount)
        If (dwErr) Then
            MsgBox("McsdGetStatus" & Chr(10) & "Error code : " & CStr(dwErr), MsgBoxStyle.OKOnly + MsgBoxStyle.Critical)
            Exit Sub
        End If
        ' カウンタは 28 ビットなので符号拡張します。
        If ((dwCount And &H8000000) = &H8000000) Then
            dwCount += &HF0000000
        End If
        MsgBox("Board status" & Chr(10) & "Axis : " & CStr(gwAxis) & Chr(10) & "Internal counter : " & CStr(dwCount) & Chr(10) & "Drive status : " & Hex(Status.bDrive) & "h" & Chr(10) & "End status : " & Hex(Status.bEnd) & "h" & Chr(10) & "Mechanical signal : " & Hex(Status.bMechanical) & "h" & Chr(10) & "Universal signal : " & Hex(Status.bUniversal) & "h", MsgBoxStyle.OKOnly + MsgBoxStyle.Information)
    End Sub
End Class